xbmc/utils/SaveFileStateJob.cpp

```diff
@@ -174,6 +173,13 @@ void CSaveFileState::DoWork(CFileItem& item,
           }
         }
 
+        //! @todo videoDBSuccess will always be true as not updated yet for play count and bookmark changes
+        if (videoDbSuccess)
+          videodatabase.CommitTransaction();
+        else
+          videodatabase.RollbackTransaction();
+        videodatabase.BeginTransaction();
+
```

Comment:
I'd only include the CommitTransaction at this time, with a todo to properly handle errors, as you did.